src/scrape.ts

```diff
@@ -27,16 +28,33 @@ const loadRailsData = async (url: string) => {
   const scriptTag = assert(
     $("script")
       .toArray()
-      .find((script) => $(script).html()?.includes("window.RailsData")),
-    "expected to find window.RailsData script tag in shopify source but couldn't"
+      .find((script) => $(script).html()?.includes("rails_data")),
+    "expected to find rails_data script in shopify source but couldn't"
   );
 
   const scriptText = $(scriptTag).html()!;
-  const jsonMatch = scriptText.match(/window\.RailsData = (\{[\s\S]*?\}\s*)\/\/\]\]/m);
+  const jsonMatch = scriptText.match(/window.__reactRouterContext.streamController.enqueue\(([\s\S]*)\)/);
   if (!jsonMatch) {
     throw new Error("RailsData object literal not found in script tag or is in an unexpected format");
   }
-  return JSON.parse(jsonMatch[1]);
+  const data = jsonMatch[1].replace("\\\n", "");
+
+  const encoder = new TextEncoderStream();
+  let controller: ReadableStreamController<any> | undefined = undefined;
+  const readableStream = new ReadableStream({
+    start(c) {
+      controller = c;
+      c.enqueue(JSON.parse(data));
+      c.close();
+    },
+  }).pipeThrough(encoder);
+
+  const result = await startDecoding(readableStream, {
+    __reactRouterContext: {
+      streamController: controller,
+    },
+  });
+  return result.value.loaderData["./routes/rest"].rails_data;
```

Comment:
Jesus -- props for figuring this out. Hacker in a past life :fedora:

Reply:
Looks like shopify actually undid their change and went back to `window.RailsData` 🤦 . I'm going to keep this here in case they revert revert